VSPackageInstaller/VSPackageInstaller/MarketplaceService/VsEditionUtil.cs

```diff
@@ -0,0 +1,112 @@
+{
+    internal static class VsSkus
+    {
+        public const string Ultimate = "Ultimate";
+        public const string Enterprise = "Enterprise";
+        public const string Pro = "Pro";
+        public const string Community = "Community";
+        public const string IntegratedShell = "IntegratedShell";
+        public const string Premium = "Premium";
+        public const string TeamExplorer = "TeamExplorer";
+        public const string SQL = "SQL";
+    }
+    static class VsEditionUtil
+    {
+        private static IDictionary<String, List<String>> EditionToSkuDictionary =
+            new Dictionary<string, List<string>>(StringComparer.OrdinalIgnoreCase)
+            {
+                {
+                    VsSkus.Enterprise, new List<string>()
+                    {
+                        VsSkus.Enterprise,
+                        VsSkus.Ultimate,
+                        VsSkus.Pro,
+                        VsSkus.Community,
+                        VsSkus.IntegratedShell
+                    }
+                },
+                {
+                    VsSkus.Ultimate, new List<string>()
+                    {
+                        VsSkus.Ultimate,
+                        VsSkus.Enterprise,
+                        VsSkus.Pro,
+                        VsSkus.Community,
+                        VsSkus.IntegratedShell
+                    }
+                },
+                {
+                    VsSkus.Premium, new List<string>()
+                    {
+                        VsSkus.Premium,
+                        VsSkus.Pro,
+                        VsSkus.Community,
+                        VsSkus.IntegratedShell
+                    }
+                },
+                {
+                    VsSkus.Pro, new List<string>()
+                    {
+                        VsSkus.Pro,
+                        VsSkus.Community,
+                        VsSkus.IntegratedShell
+                    }
+                },
+                {
+                    VsSkus.Community, new List<string>()
+                    {
+                        VsSkus.Community,
+                        VsSkus.Pro,
+                        VsSkus.IntegratedShell
+                    }
+                },
+                {
+                    VsSkus.TeamExplorer, new List<string>()
+                    {
+                        VsSkus.TeamExplorer,
+                        VsSkus.IntegratedShell
+                    }
+                },
+                {
+                    VsSkus.SQL, new List<string>()
+                    {
+                        VsSkus.TeamExplorer,
+                        VsSkus.IntegratedShell
+                    }
+                }
+            };
+
+        public static string GetCurrentVsVersion()
+        {
+            //does not give the full version value only gives 15.0
+            EnvDTE.DTE dte = ((EnvDTE.DTE)ServiceProvider.GlobalProvider.GetService(typeof(EnvDTE.DTE).GUID));
```

Comment:
This is probably fine, but as a general note, almost everything in EnvDTE is deprecated and there is probably a better way to get this information.